src/app/static/src/app/store/versions/mutations.ts

```diff
@@ -0,0 +1,31 @@
+import {MutationTree} from "vuex";
+import {VersionsState} from "./versions";
+import {PayloadWithType} from "../../types";
+import {Error} from "../../generated";
+
+export enum VersionsMutations {
+    SetManageVersions = "SetManageVersions",
+    SetLoading = "SetLoading",
+    SetSnapshotUploadPending = "SetSnapshotUploadPending",
+    VersionError = "VersionError",
+    SnapshotUploadError = "SnapshotUploadError"
+}
+
+export const mutations: MutationTree<VersionsState> = {
+    [VersionsMutations.SetManageVersions](state: VersionsState, action: PayloadWithType<boolean>) {
+        state.manageVersions = action.payload;
+    },
+    [VersionsMutations.SetLoading](state: VersionsState, action: PayloadWithType<boolean>) {
+        state.loading = action.payload;
+    },
+    [VersionsMutations.SetSnapshotUploadPending](state: VersionsState, action: PayloadWithType<boolean>) {
+        state.snapshotUploadPending = action.payload;
+    },
+    [VersionsMutations.VersionError](state: VersionsState, action: PayloadWithType<Error>) {
+        state.error = action.payload;
+        state.loading = false;
+    },
+    [VersionsMutations.SnapshotUploadError](state: VersionsState, action: PayloadWithType<Error>) {
+        console.error(action.payload);
```

Comment:
Wasn't really sure what to do here. This could happen anywhere in the stepper flow, so not really sure how to flag to user, if at all.

Reply:
We already console log all errors via the `apiService` so may as well just remove this method and have `.ignoreErrors()` set. 

Since there's no action a user can take if an auto-save fails, I guess there's not much point in advertising it but maybe we could have a little icon somewhere like "last saved: timestamp"? Which could also turn into an error icon if it fails to save?

Reply:
Good idea - I've had a stab at doing a little component for this in another branch, see what you think: https://github.com/mrc-ide/hint/pull/280